internal/service/service_integration/resource_service_integration.go

```diff
@@ -229,11 +239,22 @@ func resourceServiceIntegrationUpdate(ctx context.Context, d *schema.ResourceDat
 		return diag.FromErr(err)
 	}
 
+	userConfig := resourceServiceIntegrationUserConfigFromSchemaToAPI(d)
```

Comment:
No need for this logic here, any `array` under user configuration options follows the rules from here `convertTerraformUserConfigValueToAPICompatibleFormatArray`, and it looks like we already have this logic - sending an empty JSON list for the options that are arrays and found to be empty.